### PR TITLE
Query execution settings support

### PR DESCRIPTION
**Pull request by kburtram**

### CODE REVIEW

File: src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QuerySettingsHelper.cs [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+using Microsoft.SqlTools.ServiceLayer.QueryExecution.Contracts;
+using Microsoft.SqlTools.ServiceLayer.QueryExecution.Contracts.ExecuteRequests;
+using Microsoft.SqlTools.ServiceLayer.QueryExecution.DataStorage;
+using Microsoft.SqlTools.ServiceLayer.SqlContext;
+using Microsoft.SqlTools.ServiceLayer.Workspace;
+using Microsoft.SqlTools.ServiceLayer.Workspace.Contracts;
+using Microsoft.SqlTools.ServiceLayer.Hosting;
+using Microsoft.SqlTools.Utility;
+using System.Diagnostics;
+using System.Xml.Serialization;
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QuerySettingsHelper.cs]: Do you really need all these?

File: src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QuerySettingsHelper.cs [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+        private QueryExecutionSettings settings;
+
+        public QuerySettingsHelper(QueryExecutionSettings settings)
+        {
+            this.settings = settings;
+        }
+
+        [XmlIgnore]
+        public string SetNoCountString
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QuerySettingsHelper.cs]: Why these attributes? Are you planning on having this be serialized at all anyways?

File: src/Microsoft.SqlTools.ServiceLayer/SqlContext/QueryExecutionSettings.cs [hunk 3]
```diff
@@ -108,8 +235,389 @@ public ExecutionPlanOptions ExecutionPlanOptions
...
+        }
+
+        /// <summary>
+        /// Set transaction isolation level
+        /// </summary>
+        public string TransactionIsolationLevel
+        {
+            get
+            {
+                return GetOptionValue<string>("transactionIsolationLevel");
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/SqlContext/QueryExecutionSettings.cs]: Missing default?

**Comment by Charles-Gagnon** [comment]
>         public void Update(QueryExecutionSettings newSettings)

Add new options to this as well?

---
Refers to: src/Microsoft.SqlTools.ServiceLayer/SqlContext/QueryExecutionSettings.cs:631 in 0faff77. [](commit_id = 0faff7751540a0a952592c56ccca07848e34e315, deletion_comment = False)

File: src/Microsoft.SqlTools.ServiceLayer/Workspace/WorkspaceService.cs [hunk 4]
```diff
@@ -263,7 +265,7 @@ public void RegisterTextDocOpenCallback(TextDocOpenCallback task)
                 }
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/Workspace/WorkspaceService.cs]: Do we need to do this anymore if we're passing the URI in directly? When would openedFile be null anyways?

**Comment by Charles-Gagnon** [comment]
>             Assert.Equal(openedFile, closedFile);

Add assert for URI being the same as well?

---
Refers to: test/Microsoft.SqlTools.ServiceLayer.UnitTests/Workspace/WorkspaceTests.cs:57 in 0faff77. [](commit_id = 0faff7751540a0a952592c56ccca07848e34e315, deletion_comment = False)

**Comment by Charles-Gagnon** [comment]
>         /// <param name="scriptFile"></param>

nit: add xml doc for new param

---
Refers to: src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs:548 in 0faff77. [](commit_id = 0faff7751540a0a952592c56ccca07848e34e315, deletion_comment = False)

File: src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QuerySettingsHelper.cs [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+
+namespace Microsoft.SqlTools.ServiceLayer.QueryExecution
+{
+    /// <summary>
+    /// Service for executing queries
+    /// </summary>
+    public class QuerySettingsHelper
+    {
+        //strings for various "SET <option> ON/OFF" statements
+        private static string s_On = "ON";
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QuerySettingsHelper.cs]: nit: private static readonly

File: src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QuerySettingsHelper.cs (hunk 2 above)
Comment (kburtram) [reply src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QuerySettingsHelper.cs]: this class is copy\paste from SSMS so I left it basically as implemented there.

File: src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QuerySettingsHelper.cs (hunk 1 above)
Comment (kburtram) [reply src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QuerySettingsHelper.cs]: Done.  Copy\paste issue.  Unfortunately VS Code doesn't have "Remove unneeded using" like VS, so you have to manually figure out which ones you need.

Comment (Charles-Gagnon) [reply src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QuerySettingsHelper.cs]: I poked around a bit and turns out there is - it's just hidden under a random keybinding. ctrl + . will give you that option :

image

**Pull request merged**